runtime/tests/tests.rs

```diff
@@ -183,11 +213,64 @@ fn start_era_works() {
 	});
 }
 
+// Test to show that every extrinsic applied will add transfer fee to
+// CurrentEraTransactionRewards (until it's paid out at the end of an era)
 #[test]
-fn staking_genesis_config_works() {
+fn current_era_transaction_rewards_storage_update_works() {
 	let balance_amount = 10_000 * TransactionBaseFee::get();
-	let staked_amount = balance_amount / 5;
+	let mut total_transfer_fee: Balance = 0;
+
+	let runtime_call_1 = Call::GenericAsset(pallet_generic_asset::Call::transfer(CENTRAPAY_ASSET_ID, bob(), 123));
+	let runtime_call_2 = Call::GenericAsset(pallet_generic_asset::Call::transfer(CENTRAPAY_ASSET_ID, charlie(), 456));
+
+	ExtBuilder::default()
+		.initial_balance(balance_amount)
+		.build()
+		.execute_with(|| {
+			let xt_1 = sign(CheckedExtrinsic {
+				signed: Some((alice(), signed_extra(0, 0, None, None))),
+				function: runtime_call_1.clone(),
+			});
+			let xt_2 = sign(CheckedExtrinsic {
+				signed: Some((bob(), signed_extra(0, 0, None, None))),
+				function: runtime_call_2.clone(),
+			});
+
+			initialize_block();
+			start_era(1);
+			advance_session(); // advance a session to trigger the beginning of era 2
+
+			// Start with 0 transaction rewards
+			assert_eq!(Staking::get_current_era_transaction_fee_reward(), 0);
+
+			// Apply first extrinsic and check transaction rewards
+			assert!(Executive::apply_extrinsic(xt_1.clone()).is_ok());
+			total_transfer_fee += transfer_fee(&xt_1, TransactionPayment::next_fee_multiplier(), &runtime_call_1);
+			assert_eq!(Staking::get_current_era_transaction_fee_reward(), total_transfer_fee);
+
+			// Apply second extrinsic and check transaction rewards
+			assert!(Executive::apply_extrinsic(xt_2.clone()).is_ok());
+			total_transfer_fee += transfer_fee(&xt_2, TransactionPayment::next_fee_multiplier(), &runtime_call_2);
+			assert_eq!(Staking::get_current_era_transaction_fee_reward(), total_transfer_fee);
+
+			// Advancing sessions shouldn't change transaction rewards storage
+			advance_session();
+			assert_eq!(Staking::get_current_era_transaction_fee_reward(), total_transfer_fee);
+			advance_session();
+			assert_eq!(Staking::get_current_era_transaction_fee_reward(), total_transfer_fee);
+
+			// At the start of the next era (13th session), transaction rewards should be cleared (and paid out)
+			start_era(2);
+			advance_session();
+			assert_eq!(Staking::get_current_era_transaction_fee_reward(), 0);
```

Comment:
Would be good to check validator balances here, if possible?

Reply:
If we do check these accounts, we might need to make Dave, Eve and Ferdie do the transactions to prevent Alice, Bob and Charlie from having their accounts changed by fees.

Reply:
In my opinion, there has `balance check` before and after 1 era in `staking_genesis_config_works`, for this test is just check whether `transaction_rewards_storage` working properly.

Reply:
Discussed with @HaoZhaiX .
This is addressed in `staking_validators_should_receive_equal_transaction_fee_reward `